extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/CarMapSurfaceOwner.kt

```diff
@@ -56,7 +56,14 @@ internal class CarMapSurfaceOwner(
     logI(TAG, "surfaceAvailable")
     val oldCarMapSurface = this.mapboxCarMapSurface
     this.mapboxCarMapSurface = mapboxCarMapSurface
-    oldCarMapSurface?.let { carMapObservers.forEach { it.onDetached(oldCarMapSurface) } }
+    oldCarMapSurface?.let {
+      carMapObservers.forEach { it.onDetached(oldCarMapSurface) }
+      // The surface can become available without a call to surfaceDestroyed
+      // https://issuetracker.google.com/issues/235121269
+      oldCarMapSurface.mapSurface.onStop()
+      oldCarMapSurface.mapSurface.surfaceDestroyed()
+      oldCarMapSurface.mapSurface.onDestroy()
+    }
```

Comment:
Seems it mostly duplicates the logic at `fun surfaceDestroyed()` below, maybe extract common logic to a separate method?

Reply:
I'm not passionate either way, but my habit is the rule of three. this block also has a issue attached to it so it was not too important to share functions.

Also wondered if the function should live in the mapSurface. AA is not the only downstream implementation that will need to call onStop, surfaceDestroyed, and onDestroy?